src/pages/NewGroupPage.js

```diff
@@ -0,0 +1,272 @@
+            return sections;
+        }
+
+        sections.push({
+            title: 'RECENTS',
+            data: this.state.recentReports,
+            shouldShow: this.state.recentReports.length > 0,
+            indexOffset: sections.reduce((prev, {data}) => prev + data.length, 0),
+        });
+
+        sections.push({
+            title: 'CONTACTS',
+            data: this.state.personalDetails,
+            shouldShow: this.state.personalDetails.length > 0,
+            indexOffset: sections.reduce((prev, {data}) => prev + data.length, 0),
+        });
+
+        if (this.state.userToInvite) {
+            sections.push(({
+                undefined,
+                data: [this.state.userToInvite],
+                shouldShow: true,
+                indexOffset: 0,
+            }));
+        }
+
+        return sections;
+    }
+
+    /**
+     * Once all our options are selected this method will call the API and  create new chat between all selected users
+     * and the currently logged in user
+     */
+    createGroup() {
+        const userLogins = _.map(this.state.selectedOptions, option => option.login);
+        fetchOrCreateChatReport([this.props.session.email, ...userLogins]);
+    }
+
+    /**
+     * Removes a selected option from list if already selected. If not already selected add this option to the list.
+     * @param {Object} option
+     */
+    toggleOption(option) {
+        this.setState((prevState) => {
+            const isOptionInList = _.some(prevState.selectedOptions, selectedOption => (
+                selectedOption.login === option.login
+            ));
+
+            let newSelectedOptions;
+
+            if (isOptionInList) {
+                newSelectedOptions = _.reject(prevState.selectedOptions, selectedOption => (
+                    selectedOption.login === option.login
+                ));
+            } else {
+                newSelectedOptions = [...prevState.selectedOptions, option];
+            }
+
+            const {
+                recentReports,
+                personalDetails,
+                userToInvite,
+            } = getNewGroupOptions(
+                this.props.reports,
+                this.props.personalDetails,
+                isOptionInList ? prevState.searchValue : '',
+                newSelectedOptions,
+            );
+
+            return {
+                selectedOptions: newSelectedOptions,
+                recentReports,
+                personalDetails,
+                userToInvite,
+                searchValue: isOptionInList ? prevState.searchValue : '',
+            };
+        });
+    }
+
+    render() {
+        const maxParticipantsReached = this.state.selectedOptions.length === CONST.REPORT.MAXIMUM_PARTICIPANTS;
+        const sections = this.getSections(maxParticipantsReached);
+        const {headerTitle, headerMessage} = this.getHeaderTitleAndMessage(maxParticipantsReached);
+        return (
+            <View style={[styles.flex1, styles.p2]}>
```

Comment:
We shouldn't need `styles.p2` here, this page should be flush with its outer wrapper.

Reply:
Although we can of course address this in the follow up style PRs.

Reply:
Cool. Maybe we can save this one for when this gets moved into a modal since this is likely going to change based on that?